### DISCUSSION TITLE
Combining with UIPEthernet library

### DISCUSSION THREAD

**Discussion by HylkeWdt** [tomaskovacik/BK3254#18] [Q&A]
Hi Tomáš,

Thanks for this great library. Do you have any idea why I cannot combine the library with the UIPEthernet library?
I'm trying to create a project where via a Telnet server I can control a BK3266 and I've got both sides of the project working apart but as soon as I combine it into one there is no communication anymore from of to the BK3266. I've looked for identical named variables but could not find it.
Thanks!
Hylke

**Comment by tomaskovacik** [comment]
hello,
did including (plus initializing ? ) ethernet library into bk3254 example break functionality?

**Comment by HylkeWdt** [comment]
Hi Thomas,

 

Thanks for your fast reply,

Yes, as soon as I include and upload the UIPEthernet.h into the example cannot connect to the module anymore.

 

Best regards,

 

Hylke

 

Van: Tomáš Kováčik ***@***.***> 
Verzonden: woensdag 26 mei 2021 12:48
Aan: tomaskovacik/BK3254 ***@***.***>
CC: HylkeWdt ***@***.***>; Author ***@***.***>
Onderwerp: Re: [tomaskovacik/BK3254] Combining with UIPEthernet library (#18)

 

hello,
did including (plus initializing ? ) ethernet library into bk3254 example break functionality?

**Reply by tomaskovacik** [reply]
ok then, simple to test this on my side, stay tuned :)

**Reply by tomaskovacik** [reply]
oh this lib? https://github.com/UIPEthernet/UIPEthernet

**Reply by HylkeWdt** [reply]
Yes!

**Reply by tomaskovacik** [reply]
are u using software serial?

**Reply by tomaskovacik** [reply]
ok, I got it, problem is low memory, library is using strings which use a lot of ram, so I tested it on mega2560 and all works fine, can you test it on some chip with more ram?

**Reply by HylkeWdt** [reply]
aah, ok. Thank you for finding out! Actually I'm running the setup on a nano as it has to be build in a very small enclosure.
I think my only option then is to find another ethernet library that uses less memory.

**Reply by tomaskovacik** [reply]
there is other option, but 1st try it all out on some 1280/2560/1284, then you can strip down my library or use simply serial commands in sketch,

**Reply by HylkeWdt** [reply]
Ok, thanks for your help!
